### DISCUSSION TITLE
How to disable Banana Cake Pop (without error)?

### DISCUSSION THREAD

**Discussion by marshalluous** [ChilliCream/hotchocolate#4735] [General]
Dear Team

In your documentation you provide the information to disable Banana Cake Pop with the following code. But when I test with "https://localhost/grapqhl" it appears below error. Is this intended and Banana Cake Pop is therefore disabled or is this a bug?

Code:
`
endpoints.MapGraphQL().WithOptions(new GraphQLServerOptions
          {
              Tool = {
                  Enable = false
              }
          });
`

Error:
An unhandled exception occurred while processing the request.
AmbiguousMatchException: The request matched multiple endpoints. Matches:

Hot Chocolate GraphQL Pipeline
Hot Chocolate GraphQL Pipeline
Microsoft.AspNetCore.Routing.Matching.DefaultEndpointSelector.ReportAmbiguity(CandidateState[] candidateState)

Stack:
AmbiguousMatchException: The request matched multiple endpoints. Matches: Hot Chocolate GraphQL Pipeline Hot Chocolate GraphQL Pipeline
Microsoft.AspNetCore.Routing.Matching.DefaultEndpointSelector.ReportAmbiguity(CandidateState[] candidateState)
Microsoft.AspNetCore.Routing.Matching.DefaultEndpointSelector.ProcessFinalCandidates(HttpContext httpContext, CandidateState[] candidateState)
Microsoft.AspNetCore.Routing.Matching.DefaultEndpointSelector.Select(HttpContext httpContext, CandidateState[] candidateState)
Microsoft.AspNetCore.Routing.Matching.DfaMatcher.MatchAsync(HttpContext httpContext)
Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext)
Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.HttpsPolicy.HttpsRedirectionMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)

**Comment by tobias-tengler** [comment]
Do you have two MapGraphQL calls in your code? Or do you have two of our Map... middleware pointing at the same endpoint?
You should only have one Map... per endpoint. The endpoint is usually configured via the first method argument.

**Reply by marshalluous** [reply]
Holy Cream! You did it. I did not see the MapGraphQL at the way end of the file. Issue solved.
